tests/PHPStan/Analyser/data/bug-2648.php

```diff
@@ -37,7 +37,7 @@ public function doBar(array $list): void
 				assertType('int<0, max>', count($list));
 
 				if (count($list) === 1) {
-					assertType('int<1, max>', count($list));
```

Comment:
Do you need a testcase whether count($list) no longer is 1, when a new element is appended to $list within the IF?

Reply:
good idea, added. you made me nervous there for a second..